scene/2d/animated_sprite.cpp

```diff
@@ -362,6 +362,9 @@ void AnimatedSprite::_notification(int p_what) {
 						}
 					} else {
 						frame++;
+						// also emit if final frame
+						emit_signal(SceneStringNames::get_singleton()->frame_changed);
+						// really?
```

Comment:
because the finished is actually fired at the point the last frame is drawn, not when the loop restarts or when the animation is actually finished. I think the logic of this part should be in set_frame and _notification is calling set_frame directly.